ocf_data_sampler/numpy_sample/gsp.py

```diff
@@ -0,0 +1,33 @@
+"""Convert GSP to Numpy Sample"""
+
+import xarray as xr
+
+
+class GSPSampleKey:
+
+    gsp = 'gsp'
+    nominal_capacity_mwp = 'gsp_nominal_capacity_mwp'
+    effective_capacity_mwp = 'gsp_effective_capacity_mwp'
+    time_utc = 'gsp_time_utc'
+    t0_idx = 'gsp_t0_idx'
+    solar_azimuth = 'gsp_solar_azimuth'
+    solar_elevation = 'gsp_solar_elevation'
+    gsp_id = 'gsp_id'
+    x_osgb = 'gsp_x_osgb'
+    y_osgb = 'gsp_y_osgb'
+
+
+def convert_gsp_to_numpy_sample(da: xr.DataArray, t0_idx: int | None = None) -> dict:
```

Comment:
Not strictly necessary - but perhaps validation check for required input variables? Safe measure